### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by microshine**
- update dependencies
- move mocha config from `package.json` to `.mocahrc.yaml`
- fix type errors after dependencies update
- enable TS option `esModuleInterop`

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "incremental": true,
     "importHelpers": false,
-    "target": "ES2018",
+    "target": "ES2015",
```
Comment (WorldThirteen) [inline tsconfig.json:6]: why do we need to change target?

File: test/client/test.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
-import * as assert from "assert";
-import * as fetch from "node-fetch";
+import assert from "assert";
 import { Crypto } from "@peculiar/webcrypto";
 import { cryptoProvider } from "@peculiar/x509";
 import { ApiClient } from "@peculiar/acme-client";
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetch = require("node-fetch");
```
Comment (WorldThirteen) [inline test/client/test.ts:7]: I suppose to use node-fetch@^2.x.x rather than use require with comments at each place.

File: tsconfig.json (hunk 1 above)
Comment (microshine) [reply tsconfig.json:6]: There was an `ERR_REQUIRE_ESM` error. I forgot undo that change. Thank you

**Review by WorldThirteen — APPROVED** [review]

File: test/client/test.ts (hunk 2 above)
Comment (microshine) [reply test/client/test.ts]: It requires `@types/node-fetch` module. It's fixed now

**Pull request merged**